Release pull request for branch 20.4

robot-clickhouse opened the pull request:
This PullRequest is part of ClickHouse release cycle. It's used by CI system only. Don't perform any changes with it.

azat committed:
Access aggregate keys by names over indexes

Relax this to allow accepting aggregate keys in different order (typical
use case is distributed queries, where the initiator server will do
final merge).

alexey-milovidov committed:
Merge pull request #10761 from amosbird/llvm-libunwind

Fix libunwind

(cherry picked from commit ffd315e10f69cf9b7e5116650fc2c5408af6f5ed)

alexey-milovidov committed:
Merge pull request #10715 from CurtizJ/pk-function

Fix usage of primary key wrapped into a function.

(cherry picked from commit 63c4a7661fbe7812168d48485fd87bfa8b8dd232)

alexey-milovidov committed:
Merge pull request #10801 from ClickHouse/fix-msan-report-token

Fix (false) MSan report in MergeTreeIndexFullText

(cherry picked from commit 1aba7e45f97943957255a98b1dba6b761cde4a13)

KochetovNicolai committed:
Merge pull request #10860 from CurtizJ/pk-function

Fix sample with final (addition to #10715)

(cherry picked from commit 496d90b613bccdde0b8540448a38a643909eb1b0)

qoega committed:
Merge pull request #10908 from qoega/packager-add-bin

Allow to build binaries and tests from deb package build. 

(cherry picked from commit 2f7fd3ccfe0d13ee5d6fb04da7ce6f016dd3f3b1)

Commit:
Auto version update to [20.4.3.16] [54434]

Commit:
Auto version update to [20.4.4.1] [54434]

KochetovNicolai committed:
Merge pull request #11114 from azat/max_threads-simple-query-optimization-fix

Do not reserve extra threads after max threads optimization for simple queries

(cherry picked from commit c83014bc76629cfed0e4481957a0a5ebf2521336)

nikitamikhaylov committed:
update to clang10

nikitamikhaylov committed:
return back virtual columns

filimonov committed:
Fix for the hang during deletion of engine=Kafka (one more time) (#11145)

* Added flag for safer rdkafka destruction, but more testing detected another hang case, which has no straigt solutions and can be workarounded currenly only by draining the consumer queue, so destructor is back

* After review fixes

* After review fixes2

* After review fixes3

(cherry picked from commit 73a0ef6c0ef86acaee0c627a1e38b424555bec43)

alexey-milovidov committed:
Merge pull request #11179 from excitoon-favorites/s3globfix

Fixed S3 globbing which could fail in case of more than 1000 keys and some backends

(cherry picked from commit d4acd1af3aaef0feed94b640ba3c3b7c761ea8e8)

qoega committed:
Merge pull request #11185 from ClickHouse/fix-lambda-lc-args

Fix lambda lc args

(cherry picked from commit 2b155ac989ab5b663d91799b337117f77573331e)

Commit:
Merge pull request #11081 from vitlibar/fix-no-password-mode

Fix settings NO_PASSWORD authentication mode in users.xml.

(cherry picked from commit 2c8a355f1944004b597045bd84273fa7c7d190a0)

nikitamikhaylov commented:
Stress tests (memory) fixed in master, usban assert in lz4 fixed in master

Commit:
Auto version update to [20.4.4.18] [54434]

Commit:
Auto version update to [20.4.5.1] [54434]

KochetovNicolai committed:
Merge pull request #11200 from ClickHouse/fix-strict-resize-finishing

Fix StrictResize

(cherry picked from commit 361b92f083208f050c3a146c2f52bb2a30e89d4a)

alexey-milovidov committed:
Merge pull request #11149 from filimonov/kafka_reschedule

Fixed reschedule issue in Kafka

(cherry picked from commit 992f0dfec569e760447c9248618182fbbd20ad4e)

tavplubix committed:
Merge pull request #11205 from ClickHouse/fix-view-totals

Fix totals and extremes header for view.

(cherry picked from commit 801c1c124044b8a737136e7059b2add70e2cb1ee)

alexey-milovidov committed:
Merge pull request #11258 from vitlibar/protobuf-fix-handling-bad-data

Fix handling bad data while reading in Protobuf format.

(cherry picked from commit fb4d272f5b1b7ed7371ce166a2a33dfa6bc1f984)

alexey-milovidov committed:
Merge pull request #11255 from zlobober/proper_versions_in_arcadia

Use proper versions in Arcadia build.

(cherry picked from commit 04bdffd9d73f3f82fbbfeaae194fbd0bc45ea826)

alesapin committed:
Merge pull request #11262 from ClickHouse/fix_unit_broken_tests

Fix unit broken tests

(cherry picked from commit 9a34310ead9df490212d2b3d9253c2d79f133e6e)

nikitamikhaylov committed:
Merge pull request #11233 from nikitamikhaylov/cache-getstring-bugfix

Cache-dictionary getString() bugfix

(cherry picked from commit 9233997b24ee9b0177865a85f6b8ffb80577b9d7)

alexey-milovidov committed:
Merge pull request #11278 from vitlibar/fix-crash-set-default-role-with-wrong-args

Fix crash when SET DEFAULT ROLE is called with wrong arguments.

(cherry picked from commit 304c6a1ee38c4db78bd6481d298ec57796efd956)

alexey-milovidov committed:
Merge pull request #11303 from ClickHouse/fix-index-analysis-empty-array

Fix issue #11286; add a test

(cherry picked from commit 1dbaf80a5d7bbb2602a874925b268fb940763626)

alexey-milovidov committed:
Merge pull request #11296 from ClickHouse/fix-false-too-slow

Fix bug with Throttler and query speed estimation

(cherry picked from commit 97d9f218af06a97d6810ab5d7594602f7c7055f5)

alexey-milovidov committed:
Merge pull request #11314 from ClickHouse/fix-race-condition-thread-pool

Fix very rare race condition in ThreadPool

(cherry picked from commit 1df5c28dbc0fe0ea9cbab04c58aee00147af3276)

alexey-milovidov committed:
Merge pull request #11313 from ClickHouse/fix-race-condition-cluster-copier

Fix data race in clickhouse-copier

(cherry picked from commit 3de876b2bc1c937f7b60464fd396a756532957cd)

alexey-milovidov committed:
Merge pull request #11311 from ClickHouse/fix-uninitialized-memory-in-cast

Fix uninitialized memory in conversion

(cherry picked from commit 1a8c19741e407e478b5d4c7f980c30c235ca98bc)

alexey-milovidov committed:
Merge pull request #11318 from devwout/master

Fix visitParamExtractRaw for strings with unbalanced { or [.

(cherry picked from commit 5b2b3739b9ae9be97d90afba0cf00f2d6a4c70c4)

alexey-milovidov committed:
Merge pull request #10621 from azat/enable_optimize_predicate_expression-order

Fix predicates optimization for distributed queries with HAVING

(cherry picked from commit a8f5b10f156fccbf4801879e67c0c865df0ba9e3)

alexey-milovidov committed:
Merge pull request #11162 from azat/data-skip-index-merging-params-fix

[RFC] Fix data skipping indexes for columns with additional actions during merge

(cherry picked from commit 8accde79b522dc618e4096c8ab2050dfe07f86e8)

4ertus2 committed:
fix select from StorageJoin

4ertus2 committed:
make clang-10 happy

alexey-milovidov committed:
Merge pull request #11337 from ClickHouse/fix-quantile-arguments

Fix quantile arguments

(cherry picked from commit 68e2f88635c3f3b1fe6aca81db986fc2413a26fd)

alexey-milovidov committed:
Merge pull request #11330 from ClickHouse/fix-10241

More parallel MV processing.

(cherry picked from commit 92ac608447289ef5fd484b0dd06a813afd323b1a)

qoega committed:
Merge pull request #11335 from ClickHouse/stop_merges_before_alter

Stop merges before taking exclusive lock :(

(cherry picked from commit a101b17df72ba23f9772f64d56cb616f9902ea27)

alexey-milovidov committed:
Merge pull request #11348 from ClickHouse/lazy-database-segfault-2

Fix potential segfault when using Lazy database.

(cherry picked from commit 9b07989954f7c986d1e410d5d1d2a9f683a6a5ae)

alexey-milovidov committed:
Merge pull request #11354 from ClickHouse/get_rid_of_lock_during_send

Remove structure lock on parts send

(cherry picked from commit ff7262968a5f546496969ea3cba5813a9c05ea5b)

4ertus2 committed:
better errors for joinGet()

alexey-milovidov committed:
Merge pull request #11350 from ClickHouse/fix-client-vertical-multiline

Fix \G in clickhouse-client multiline mode

(cherry picked from commit 7ec11cad659ce24709e2a7c0a6d62ddd622deddb)

alesapin committed:
Manual cherry-pick of #11355

akuzm committed:
Merge pull request #11366 from ClickHouse/fix-SortingTransform-bug

Fix sorting transform bug

(cherry picked from commit 8bf8b44510235cbd72c3aa59f47b227ac31b75bb)

4ertus2 committed:
better error for USING key

4ertus2 committed:
fix join over LC and Nullable in join key

KochetovNicolai committed:
Added test.

4ertus2 committed:
less optimistic patch

alexey-milovidov committed:
Merge pull request #11403 from Enmk/geohashesInBox-fix

Fixed geohashesInBox argument range

(cherry picked from commit acf77739424fac49f8de4b40da0400a8647e2b27)

alesapin committed:
Merge pull request #11420 from ClickHouse/fix-msan-failure

Fix MSan failure in MergeTree background task

(cherry picked from commit 0a453f5c8335d638ed1137794bdbae982e5a2823)

alesapin committed:
Merge pull request #11431 from ClickHouse/fix_codecs_arguments_crash

Fix segfault with wrong codecs arguments

(cherry picked from commit 603917999be33a0067bf174b316f74a9eb7fcf50)

alexey-milovidov committed:
Merge pull request #11448 from ClickHouse/fix-codecs-compression-size

Try fix return compressed size for codecs.

(cherry picked from commit 0d30d1005f397959cd5600b43f5614f060ff1ba5)

alexey-milovidov committed:
Merge pull request #11421 from livace/fix_point_in_polygon_with_nan

Fix pointInPolygon with nan

(cherry picked from commit 42534959e345ec720bc5fad29d36c4d0652491dc)

KochetovNicolai committed:
Fix build.

nikitamikhaylov committed:
fix build

alexey-milovidov committed:
Merge pull request #11496 from ClickHouse/fix-aggregation-state-exception-safety-memory-leak

Fix memory leak when exception is thrown in the middle of aggregation with -State functions

(cherry picked from commit 2bff9b4607126ec408c1095003eb8bfb2bac1f7a)

alexey-milovidov committed:
Merge pull request #11523 from ClickHouse/fixing_01079_tests

Fixing test and benign race condition during table drop

(cherry picked from commit 91d06c7e4f41cd9783c7add2a83573514345b0bd)

alesapin committed:
Fix autogenerated_versions.txt

KochetovNicolai committed:
Empty commit.

alexey-milovidov committed:
Merge pull request #11540 from ClickHouse/fix-direct-io

Fix direct IO with PREWHERE

(cherry picked from commit 8e05558a5fb8d8639d1638361621d0fa090a3b13)

alexey-milovidov committed:
Merge pull request #11550 from realadvisor/master

Fix ClickHouse init process failed in docker if any script is added into docker-entrypoint-initdb.d 

(cherry picked from commit 13b225799ec4a136dcfd08b1d5209be3cc7dba55)

KochetovNicolai commented:
Only `test_default_role` integration tests failed.

Commit:
Auto version update to [20.4.5.36] [54434]

Commit:
Auto version update to [20.4.6.1] [54434]

alexey-milovidov committed:
Merge pull request #11528 from azat/use_compact_format_in_distributed_parts_names-fix

Fix shard_num/replica_num for <node> (breaks use_compact_format_in_distributed_parts_names)

(cherry picked from commit 123a0170572f69706ee14bf68f5a9a6ef82f3e12)

alexey-milovidov committed:
Merge pull request #11589 from ClickHouse/fix-log-message

Fix trivial error in log message #11399

(cherry picked from commit 0aef84605aa968426f78da7f4122c9e348258491)

alexey-milovidov committed:
Merge pull request #11580 from ClickHouse/fix-bug-prewhere-in-subquery

Fix prewhere in subquery

(cherry picked from commit 9e688307cb61edb659dd74373ba5e47f2db15a48)

alexey-milovidov committed:
Merge pull request #11579 from ClickHouse/fix_11490

Fix segfault in SHOW CREATE TABLE

(cherry picked from commit 9c20afbc8191fd9c9e0896d644c0db8c24b83e47)

alexey-milovidov committed:
Merge pull request #11578 from ClickHouse/fix_10240

Fix query_id of http queries

(cherry picked from commit c7be0d6bc09f9ca12991b1f4a5d8ea93471b88f4)

KochetovNicolai committed:
Added test from #11259.

KochetovNicolai committed:
Revert "Merge pull request #11162 from azat/data-skip-index-merging-params-fix"

This reverts commit d7ba41db48b35cf5130b931c7d392d2d7383ba97.

alesapin committed:
Fix for fix from #11259 and #11162

azat commented on src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp:
```diff
@@ -794,6 +794,12 @@ MergeTreeData::MutableDataPartPtr MergeTreeDataMergerMutator::mergePartsToTempor
+    {
+        merged_stream = std::make_shared<ExpressionBlockInputStream>(merged_stream, data.skip_indices_expr);
+        merged_stream = std::make_shared<MaterializingBlockInputStream>(merged_stream);
+    }
```
But #11162 has been already reverted from 20.4 in d2ee7dc0f2ec617ff6dbec7658964d3e6e5368f3, this means that there is no test for this change right now, can you pick the test too then?

alexey-milovidov committed:
Merge pull request #11601 from filimonov/fix_client_exit_code

Fix corner case (only) for exit code overflow

(cherry picked from commit d5c2008c7d1ae114205848f68de8faec6e0e6858)

alexey-milovidov committed:
Merge pull request #11604 from giordyb/patch-1

Update entrypoint.sh

(cherry picked from commit 1465caa325d21589e6d8a2e4397cb853ecaca1ed)

alexey-milovidov committed:
Merge pull request #11608 from ClickHouse/fix-nullable-prewhere-type

Fix nullable prewhere type

(cherry picked from commit f53da4d36b8c3a214567c935caed478edce08363)

alexey-milovidov committed:
Merge pull request #11608 from ClickHouse/fix-nullable-prewhere-type

Fix nullable prewhere type

(cherry picked from commit f53da4d36b8c3a214567c935caed478edce08363)

alexey-milovidov committed:
Merge pull request #11644 from ClickHouse/remove-trivial-count-optimization-for-row-level-security

Remove trivial count query optimization if row-level security is set

(cherry picked from commit 47316dd59d0f4cbc570252d3a016bb22cb594c2e)

alexey-milovidov committed:
Merge pull request #11649 from ClickHouse/case-insensitive-regexp

Allow case-insensitive regexps; added a test

(cherry picked from commit b9b725a39f8f78c8378fc2aa6ea96f99420adba4)

alexey-milovidov committed:
Merge pull request #11653 from ClickHouse/parse-date-time-best-effort-incompletely-parsed-value

Return NULL or zero when value is not parsed completely in parseDateTimeBestEffortOr* functions

(cherry picked from commit 88cf3f708309d0e8581afae91bab86316330c973)

akuzm committed:
Merge pull request #11638 from azat/skip-idx-bloom-filter-fix

Fix bloom filters for String (data skipping indices)

(cherry picked from commit e460d7c2bd58b379cc68a2a2e144305af36f9c02)

alexey-milovidov committed:
Merge pull request #11689 from CurtizJ/fix-with-ties

Fix 'LIMIT WITH TIES' with aliases

(cherry picked from commit 82b27ab618c731eb7b03608b433f6bd91fefcff5)

Commit:
Merge pull request #11671 from vitlibar/fix-minor-bugs-in-rbac

Fix minor bugs in RBAC

(cherry picked from commit c5229301fc5404423870d237d9fdf4882efc23dc)

alesapin committed:
Merge pull request #11681 from ClickHouse/fix_mutations_premature_finish

Fix mutations premature "finish"

(cherry picked from commit d214a504d7ae525ab62560971669713a2271c287)

alexey-milovidov committed:
Merge pull request #11690 from ClickHouse/odbc-bridge-timeouts

Pass proper timeouts when communicating with XDBC bridge.

(cherry picked from commit 6e61f708d794536b49be97655b6725067acbb9f8)

nikitamikhaylov committed:
Merge pull request #11602 from ClickHouse/fix-buffer-sample

Fix buffer sample

(cherry picked from commit d2be6c036b3b2f4bec8de985a9c620f9ae2cd28d)

tavplubix committed:
Merge pull request #11732 from ClickHouse/fix_type_duduction_in_values_format

Fix type duduction in Values format

(cherry picked from commit bb16544bee0faeeeb82c2d68c5d2be566b4b8666)

alexey-milovidov committed:
Merge pull request #11697 from CurtizJ/fix-with-fill

Fix 'ORDER BY ... WITH FILL' over const columns.

(cherry picked from commit c9924b8aa16f7222f8db52bcef08e0b38a3884d6)

alesapin committed:
Merge pull request #11744 from ClickHouse/stop_fetches_before_metadata_alter

Cancel fetches before metadata alter

(cherry picked from commit d0beeaf7c4bf98c2dd392beb283c87107e0363d5)

alexey-milovidov committed:
Merge pull request #11788 from ClickHouse/fix-too-many-threads

Fix too many threads

(cherry picked from commit 4129300fdb6f7a1bfea1b4796ca053c556562438)

alexey-milovidov committed:
Merge pull request #11828 from ClickHouse/fix-fixed-string-comparison

Fix FixedString comparison #11393

(cherry picked from commit 485c4c9055c358e7b636e899276100504770159f)

alexey-milovidov committed:
Merge pull request #11846 from ClickHouse/prohibit-array-join-in-higher-order-functions

Don't allow arrayJoin inside higher order functions

(cherry picked from commit dd31bcff6664805b145f7d2be7c64a02ffc34f3e)

Commit:
Fix integration test test_default_role.

Commit:
Revert "Merge pull request #11608 from ClickHouse/fix-nullable-prewhere-type"

This reverts commit 592c9d2f73f26996945860929b05bc145686dcf6.

Commit:
Fix compilation error: could not include "hs.h"

alexey-milovidov committed:
Merge pull request #11869 from ClickHouse/fix-nullable-prewhere-type-2

Fix nullable prewhere type 2

(cherry picked from commit a7b224a57837553cd8c09b3411b39e89e3fac5be)

alexey-milovidov committed:
Merge pull request #11840 from azat/fix-http-memory-accounting

Fix memory accounting via HTTP interface

(cherry picked from commit b4475086abbceff19401bbc5b99b1716934b2827)

hexiaoting committed:
fix bug for query "select *, abc.*" should failed. (#11753)


(cherry picked from commit 9135cb1bf1dac7318b880c09a316a261f991895e)

KochetovNicolai committed:
Fix anotehr one nullable prewhere column.

(cherry picked from commit 658a2d0dc010154c1cb486ea044b0135fe366512)

4ertus2 committed:
Fix if function with NULLs (#11807)


(cherry picked from commit 92016e8bfb3018b318b5522b21c02e3e58ad2f64)

alexey-milovidov committed:
Merge pull request #11875 from ClickHouse/fix-fpe-datetime64

Fix strange and wrong code around DateTime64

(cherry picked from commit c2fba5179bca2fa7a83f36dd85e7f748d1579637)

Commit:
Fix using current database while checking access rights.

(cherry picked from commit 29a9d4187f569cc781cffbc712c1a3ddb3fcfb82)

alexey-milovidov committed:
Merge pull request #11916 from ClickHouse/query-parameters-keep-alias

Keep aliases for substitutions in query.

(cherry picked from commit 2ce2588770e4080d6b739f9542a872e6c70136b3)

Commit:
Auto version update to [20.4.6.53] [54434]

Commit:
Auto version update to [20.4.7.1] [54434]

tavplubix committed:
Merge pull request #11934 from ClickHouse/fix_some_logical_errors

Fix some logical errors

(cherry picked from commit 0260358253c5240f545b31d995250eaf039aecc9)

alexey-milovidov committed:
Merge pull request #11936 from ClickHouse/fix_11918

Fix query parameter visitor

(cherry picked from commit 6a45f3e960f0a8729b4dae8f2b5e8259f2a23ab7)

alexey-milovidov committed:
Merge pull request #11997 from ClickHouse/fix-bad-log-message

Fix bad log message at server startup

(cherry picked from commit 1d7d5e74f6454b93b69c9f5cfa6c7a31f58f70cb)

alexey-milovidov committed:
Merge pull request #12005 from ClickHouse/fix-div-nontrivial-fpe

Fix potential FPE.

(cherry picked from commit d867dd8a65a9fbad44a3757e5793e2a2eefbda71)

alexey-milovidov committed:
Merge pull request #12013 from ClickHouse/fix-array-size-overflow-generate-random

Fix array size overflow in generateRandom

(cherry picked from commit 95d31d755a339ce31188edad32d42811a7e2cfd8)

Commit:
Merge pull request #12015 from vitlibar/fix-access-rights-allow-ddl-0

Fix calculation of access rights when allow_ddl = 0

(cherry picked from commit f3f005d5b9ac6cef89f7249cb1fc50da1fd4b0b1)

akuzm committed:
Merge pull request #12019 from ClickHouse/fix-neighbor-overflow

Fix overflow in function "neighbor"

(cherry picked from commit 32237578086785b05c8ce456e92c1f58ccb2b671)

alexey-milovidov committed:
Merge pull request #12039 from ClickHouse/fix-nullable-tuple-compare

Fix nullable tuple compare

(cherry picked from commit c1d2d2d7f759536cdac991077110ea40023030c2)

alexey-milovidov committed:
Merge pull request #12043 from ClickHouse/topk-enum

Make topK return Enum for Enum types

(cherry picked from commit 08b6ca19d8fece2680b5a7e7ec43eea914405de6)

Commit:
Merge pull request #12031 from vitlibar/fix-access-rights-if-allow_introspection_functions-is-0

Fix calculation of access rights when allow_introspection_functions = 0

(cherry picked from commit 40942a2164e7d0d4fef55460348906d87e5a2d32)

alexey-milovidov committed:
Merge pull request #12045 from ClickHouse/fix_loading_of_tables

Parse metadata in parallel when loading tables

(cherry picked from commit 948af0bfc51c3a56207b7cd4bfcee356b6216080)

Akazz committed:
Merge pull request #11893 from excitoon-favorites/autostartmoves

In *MergeTree: Parts moving task shall be started if new storage policy needs them

(cherry picked from commit cc2cadb7f529f90277a47e52c7cede91241ee9c6)

alesapin committed:
Merge pull request #12056 from ClickHouse/fix-11808

Fix defaultValueOfArgumentType

(cherry picked from commit 65dae99c125e5fef92b96770916a85cee0252941)

alesapin committed:
Merge pull request #12055 from ClickHouse/fix-9766

Fix arrayMap with Tuple(LowCardinality) argument

(cherry picked from commit f87c3af09778e2da57b4c7dc0af700dcc69f39a8)

blinkov committed:
[docker] install ca-certificates before the first apt-get update (#12095)

* [docker] install ca-certificates before first apt-get update

* Update Dockerfile

(cherry picked from commit b81c66a03148afc02647df3c11aec40382329912)

KochetovNicolai committed:
Merge pull request #12085 from ClickHouse/fix-11937

Fix limiting the number of threads for VIEW.

(cherry picked from commit 17c89f3fe0f85e293fb5a5ec907491068c1ab52d)

alesapin committed:
Fix images.json

alexey-milovidov committed:
Merge pull request #12133 from ClickHouse/normalize-pid-file

Normalize "pid" file handling #3501

(cherry picked from commit f1151ed4742cb812b60c7080a607d008ba24c8e4)

alexey-milovidov committed:
Merge pull request #12089 from ClickHouse/fix-11595

Fix result_rows and result_bytes metrics for selects.

(cherry picked from commit cf1967bb29654a9a46e70b6f166b1666f821a415)

alexey-milovidov committed:
Merge pull request #12060 from ClickHouse/fix-12053

Check wrong type for filter.

(cherry picked from commit e05283d5d3c8cc1f9a5bf1a78a0fd4be915f97e7)

alexey-milovidov committed:
Merge pull request #12137 from ClickHouse/fix-loop-geodistance

Remove harmful code from "geoDistance"

(cherry picked from commit f02edaa48c02e9e508ac06a1f195d2b405ce38de)

alexey-milovidov committed:
Merge pull request #12140 from ClickHouse/fix-division

Fix potential overflow in integer division

(cherry picked from commit eecf7ae3eea26a188209ceac51f708f0ba329f53)

alexey-milovidov committed:
Merge pull request #12042 from ClickHouse/fix-constraints

Fix constraints check for constant columns, step 1.

(cherry picked from commit 0f1bfa5b190f1068fa479caa2f38b88f7d6bd700)

alexey-milovidov committed:
Merge pull request #12021 from ClickHouse/fix-if-fixed-string

Fix function if with FixedString arguments of different sizes

(cherry picked from commit ab15c8d6cab74715b3fe94f93774ff6615672549)

alexey-milovidov committed:
Merge pull request #12116 from vitlibar/fix-table-dictionary-dependency

Fix handling dictionary-table dependency

(cherry picked from commit ea0fb005fb1766b526ffdb29275b2f6e8cbbb819)

alexey-milovidov committed:
Merge pull request #12092 from CurtizJ/fix-state-resample

Fix segfault with -StateResample combinators

(cherry picked from commit 3aad1221533ab8498a78020ec907afe13178dcf4)

alesapin committed:
Merge pull request #12153 from ClickHouse/fix-mutations-interpreter

Fix wrong logic in MutationsInterpreter

(cherry picked from commit 8e767bf5d098eb9170393a4f9545731efafd735d)

alesapin committed:
Merge pull request #12165 from vitlibar/fix-split-dictionary-source-table-name

Fix splitting table name of dictionary source

(cherry picked from commit be85f9f98ab3d9c5450be967f7fed646d0ce3129)

alexey-milovidov committed:
Merge pull request #12151 from ClickHouse/fix-transform-query-aliases

Fix transform query for external databases in presense of aliases

(cherry picked from commit 546fcc9ed6cb6e36e89a4122ff0dc696e6b5af71)

alexey-milovidov committed:
Merge pull request #12196 from ClickHouse/akz/logical-functions-normalized-uint8

Fixed logical functions for UInt8 values when they are not equal to 0 or 1.

(cherry picked from commit 0fdc29be1eef2c67c973cff531455116fa17326d)

alexey-milovidov committed:
Merge pull request #12271 from ClickHouse/fix-logical-functions-review

Fix minor issues in ternary logic

(cherry picked from commit d0d1590d0d2a13dd07b875181c8ea5c3adad2369)

alexey-milovidov committed:
Merge pull request #12275 from ClickHouse/supertype-low-cardinality

Supertype for LowCardinality

(cherry picked from commit 3f0c2573899c3181104f911a9e8308c24242eb93)

alexey-milovidov committed:
Merge pull request #12279 from ClickHouse/fix-array-fill

Fix arrayFill for empty arrays.

(cherry picked from commit df8cde7b546f921a02c9290fd70e9e3d60f399c2)

tavplubix committed:
Merge pull request #12197 from nikitamikhaylov/csv-file-bugfix

Headers for CSVWithNames in StorageFile

(cherry picked from commit 33c3545125768fc78b5f1a126daf7359d23a9d33)

Commit:
Merge pull request #12290 from vitlibar/show-error-after-trie-dictionary-failed-to-load

Show error after TrieDictionary failed to load.

(cherry picked from commit d3230e5566c141047a283416849c1eb03c24b2e2)

alexey-milovidov committed:
Merge pull request #12103 from ClickHouse/fix-12030

Fix over-limiting the number of threads for union.

(cherry picked from commit cbb539c682b52d9844f43251cbe61ace74bf862c)

alexey-milovidov committed:
Merge pull request #12179 from azat/GROUP-BY-injective-elimination-dictGet-fixes

Fix dictGet arguments check during GROUP BY injective functions elimination

(cherry picked from commit 0c37fe9c7573d8add09ddd5987e03c1dfbe56ac4)

alexey-milovidov committed:
Merge pull request #12305 from ClickHouse/fix-virtual-columns-filter

Fix virtual columns filter

(cherry picked from commit 827990d6816028d4cd9a40835b7801da48aedf4a)

CurtizJ committed:
Merge pull request #12306 from CurtizJ/fix-with-fill

Fix order of columns in WITH FILL modifier

(cherry picked from commit c4767557f24dc4530dd2e8e1991115242743716e)

alesapin committed:
Fix exceptions

alesapin committed:
Fix exceptions handling

alesapin committed:
Fix exception

alesapin committed:
merge fixes

filimonov committed:
Backport #12120 (#12396)

Add missing query context for system logs

Needed to allow attaching materialized views with joins or with
subqueries to system.logs.

Fix #10437, CR fixes

Sync reference file with changes in sql file

Same change for Kafka - just in case, and to make it conform.

alesapin committed:
Fix

alesapin committed:
Fix exceptions

alexey-milovidov committed:
Merge pull request #12401 from ClickHouse/fix_segfault_in_storage_merge

Fix another Context-related segfault

(cherry picked from commit 7c2bd32c9c9d3cf7f0e7c5a777d43760410027de)

alexey-milovidov committed:
Merge pull request #12400 from vitlibar/fix-bad_typeid

Fix std::bad_typeid when JSON functions called with argument of wrong type

(cherry picked from commit c615ea658bd9bad1165260db7fd5c9222e46bd93)

alesapin committed:
Fix 01373 test

alesapin committed:
Disable test 01360_materialized_view because it require improvement

(cherry picked from commit df1b7fd8e72ce2b015d31ebc425863a7d538ba1d)

alesapin committed:
Fix test 01091_num_threads

alesapin committed:
Fixed test 01358_constexpr_constraint

alesapin committed:
Fix odbc bridge

alesapin committed:
Fix test test_dictionaries_dependency

alesapin committed:
Suppress settings profiles

alesapin commented:
All important tests should be fixed, but flaky failures still may happen.

alesapin commented:
Memory stress:
```
2020.07.14 08:33:10.377725 [ 51650 ] {} <Fatal> BaseDaemon: ########################################
2020.07.14 08:33:10.377864 [ 51650 ] {} <Fatal> BaseDaemon: (version 20.4.7.67 (official build)) (from thread 4789) (query_id: 16f55d4c-af63-4a84-8be1-6179741904b5) Received signal Aborted (6).
2020.07.14 08:33:10.377943 [ 51650 ] {} <Fatal> BaseDaemon: 
2020.07.14 08:33:10.378020 [ 51650 ] {} <Fatal> BaseDaemon: Stack trace: 0x7fd879c473eb 0x7fd879c26899 0x9827797 0x98261c1 0x97c9f3f
2020.07.14 08:33:10.378188 [ 51650 ] {} <Fatal> BaseDaemon: 4. gsignal @ 0x463eb in /usr/lib/x86_64-linux-gnu/libc-2.30.so
2020.07.14 08:33:10.378277 [ 51650 ] {} <Fatal> BaseDaemon: 5. abort @ 0x25899 in /usr/lib/x86_64-linux-gnu/libc-2.30.so
2020.07.14 08:33:10.382322 [ 51650 ] {} <Fatal> BaseDaemon: 6. ? @ 0x9827797 in /usr/bin/clickhouse
2020.07.14 08:33:10.386267 [ 51650 ] {} <Fatal> BaseDaemon: 7. ? @ 0x98261c1 in /usr/bin/clickhouse
2020.07.14 08:33:10.390218 [ 51650 ] {} <Fatal> BaseDaemon: 8. ? @ 0x97c9f3f in /usr/bin/clickhouse
```
Seems like we don't have some changes for memory sanitizer?

alesapin commented:
ubsan 
```../contrib/lz4/lib/lz4.c:818:44: runtime error: applying zero offset to null pointer```
Missing a patch in lz4?

alesapin commented:
DatabaseAtomic failures not important, experimental functionality.

alesapin commented:
Functional tests with memory sanitizer -- `01081_demangle`. Strange error, but it works in the release build, so skipsit.

alesapin commented:
Stress tests with address sanitizer:
```
==273==ERROR: AddressSanitizer: allocator is out of memory trying to allocate 0x20000 bytes
==273==ERROR: AddressSanitizer failed to allocate 0x2000 (8192) bytes of InternalMmapVector (error code: 12)
```
It's ok, we have fixed this error in master, but I don't remember where and when.

alesapin commented:
ClickHouse Unit Tests Msan -- infinite recursion in libunwind.

alesapin commented:
No integration tests. Run them manually.

alesapin commented:
Asan has known failure
```
==273==ERROR: AddressSanitizer: allocator is out of memory trying to allocate 0x20000 bytes
==273==ERROR: AddressSanitizer failed to allocate 0x2000 (8192) bytes of InternalMmapVector (error code: 12)
ERROR: Failed to mmap```

alesapin commented:
`test_moves_to_disk_work` no fix in test in this branch. It's ok.

Commit:
Auto version update to [20.4.7.67] [54434]

Commit:
Auto version update to [20.4.8.1] [54434]

4ertus2 committed:
Fix crash in 'JOIN dict ON expr(dict_key) = k' (#12458)


(cherry picked from commit e8057387cbdc6e9990eef805f8f103871708ad30)

alexey-milovidov committed:
Merge pull request #12529 from CurtizJ/fix-12521

Better exception for function 'in' with invalid number of arguments

(cherry picked from commit 6e9af5f17919b17eca7c6ad10b689233e4204432)

alesapin committed:
Merge pull request #12543 from ClickHouse/fix_marks_type_on_mutation

Fix incorrect adaptive granularity by default

(cherry picked from commit ac802a8e9654224a620a1ff1fce78628b3f4c9c2)

tavplubix committed:
Merge pull request #12556 from azat/distributed-remove-files-on-drop

Remove data for Distributed tables (blocks from async INSERTs) on DROP TABLE

(cherry picked from commit 5634482393c502d116b3dcccc5ade9880d9211a6)

CurtizJ committed:
fix build

CurtizJ committed:
Revert "Merge pull request #12556 from azat/distributed-remove-files-on-drop"

This reverts commit 3c2fff8649d2c1f09e1006e7a09f5fb652f5fcba.

alexey-milovidov committed:
Merge pull request #12566 from ClickHouse/race_condition_cache_dictionaries

Fix race condition in clickhouse cache dictionary

(cherry picked from commit 16e8c6143847e654e51fadda8ba1e83fc1d28cba)

KochetovNicolai committed:
Added test with extra overflow row.

zhang2014 committed:
ISSUES-12293 allow push predicate when subquery contains with clause (#12663)

KochetovNicolai committed:
Add lost aggregating info.

KochetovNicolai committed:
Merge pull request #12747 from ClickHouse/fix-extra-overflow-row

Fix extra overflow row

Commit:
Backport #12659 to 20.4: ISSUES-10572 fix bloom filter with const column

Commit:
Backport #12672 to 20.4: Fix memory tracking for input_format_parallel_parsing

alexey-milovidov committed:
Merge pull request #12859 from ClickHouse/backport/20.4/12659

Backport #12659 to 20.4: ISSUES-10572 fix bloom filter with const column

alexey-milovidov committed:
Merge pull request #12865 from ClickHouse/backport/20.4/12672

Backport #12672 to 20.4: Fix memory tracking for input_format_parallel_parsing

Commit:
Backport #12648 to 20.4: Add support for function if with Array(UUID) arguments

Commit:
Backport #12633 to 20.4: Fix \"There is no supertype\" error on ALTER UPDATE #7306

Commit:
Backport #12182 to 20.4: Cap max_memory_usage* limits to the process resident memory

Commit:
Backport #12062 to 20.4: Try fix pk in tuple performance

Commit:
Backport #11739 to 20.4: [RFC] Parse metadata stored in zookeeper before checking for equality

CurtizJ committed:
Merge pull request #12928 from ClickHouse/backport/20.4/12062

Backport #12062 to 20.4: Try fix pk in tuple performance

alexey-milovidov committed:
Merge pull request #12914 from ClickHouse/backport/20.4/12182

Backport #12182 to 20.4: Cap max_memory_usage* limits to the process resident memory

alexey-milovidov committed:
Merge pull request #12935 from ClickHouse/backport/20.4/11739

Backport #11739 to 20.4: [RFC] Parse metadata stored in zookeeper before checking for equality

alexey-milovidov committed:
Merge pull request #12881 from ClickHouse/backport/20.4/12633

Backport #12633 to 20.4: Fix "There is no supertype" error on ALTER UPDATE #7306

alexey-milovidov committed:
Merge pull request #12878 from ClickHouse/backport/20.4/12648

Backport #12648 to 20.4: Add support for function if with Array(UUID) arguments

Commit:
Backport #12658 to 20.4: Fix SIGSEGV in StorageKafka when broker is unavailable

alexey-milovidov committed:
Merge pull request #12970 from ClickHouse/backport/20.4/12658

Backport #12658 to 20.4: Fix SIGSEGV in StorageKafka when broker is unavailable

KochetovNicolai committed:
Merge pull request #12840 from ClickHouse/remove-wrong-threads-limiting

Remove thread limiting.

(cherry picked from commit 1f2390b65e921167dab6249ad9fa1ec29ab9bc91)

robot-clickhouse committed:
Backport #12625 to 20.4: Better exception during directories creation (#12875)

robot-clickhouse committed:
Backport #12613 to 20.4: Remove sort description from streams (#12872)

robot-clickhouse committed:
Backport #12618 to 20.4: Fix external sort pipeline stuck (#12884)

Commit:
Backport #12492 to 20.4: Make read buffer size lower, while reading from compact parts

CurtizJ committed:
Merge pull request #13005 from ClickHouse/backport/20.4/12492

Backport #12492 to 20.4: Make read buffer size lower, while reading from compact parts

robot-clickhouse committed:
Backport #12989 to 20.4: Fix `Block structure mismatch` error for queries with  UNION and JOIN (#13048)

robot-clickhouse committed:
Backport #12982 to 20.4: Merging #12548 - Correction to `merge_with_ttl_timeout` logic by @excitoon (#13064)

Commit:
Backport #13081 to 20.4: Fix wrong index analysis with functions

alexey-milovidov committed:
Merge pull request #13145 from ClickHouse/backport/20.4/13081

Backport #13081 to 20.4: Fix wrong index analysis with functions

Commit:
Backport #12700 to 20.4: Fix performance with large tuples

alexey-milovidov committed:
Merge pull request #13186 from ClickHouse/backport/20.4/12700

Backport #12700 to 20.4: Fix performance with large tuples

abyss7 committed:
Revert "Backport #11739 to 20.4: [RFC] Parse metadata stored in zookeeper before checking for equality"

This reverts commit e86cfcd37796b9ca339f99d46503d29ff97cfebb.

abyss7 committed:
Fix build

Commit:
Backport #13224 to 20.4: Fix assert in arrayElement

alexey-milovidov committed:
Merge pull request #13247 from ClickHouse/backport/20.4/13224

Backport #13224 to 20.4: Fix assert in arrayElement

Commit:
Backport #13226 to 20.4: Fix function if with nullable constexpr as cond that is not literal NULL

robot-clickhouse committed:
Backport #12035 to 20.4: Fix crash with low card column in merge join (#13008)

robot-clickhouse committed:
Backport #12857 to 20.4: Fix columns duplication for range hashed dictionary (DDL) (#12986)

alexey-milovidov committed:
Merge pull request #13265 from ClickHouse/backport/20.4/13226

Backport #13226 to 20.4: Fix function if with nullable constexpr as cond that is not literal NULL

Commit:
Backport #13254 to 20.4: Fix hash table with NaN keys

alexey-milovidov committed:
Merge pull request #13296 from ClickHouse/backport/20.4/13254

Backport #13254 to 20.4: Fix hash table with NaN keys

alesapin committed:
Bump CI

Commit:
Backport #13234 to 20.4: Return passed number for numbers with MSB set in roundUpToPowerOfTwoOrZero()

robot-clickhouse committed:
Backport #13386 to 20.4: Deadlock in textlog (#13414)

Commit:
Backport #13441 to 20.4: Fix error in parseDateTimeBestEffort

akuzm committed:
update mysql odbc driver version in integration test docker

Commit:
Backport #13396 to 20.4: Fix assertion in FinishSortingTransform.

Commit:
Backport #13420 to 20.4: Fix tuple with null comparison

alesapin committed:
Merge pull request #13494 from ClickHouse/fix_deb_builder

New version of dpkg-deb in deb builder.

(cherry picked from commit 91fec04cdd16d44ec4c10c9a31669c173c6098c5)

alexey-milovidov committed:
Merge pull request #13464 from ClickHouse/backport/20.4/13441

Backport #13441 to 20.4: Fix error in parseDateTimeBestEffort

Commit:
Auto version update to [20.4.8.99] [54434]

Commit:
Auto version update to [20.4.9.1] [54434]

alexey-milovidov committed:
Merge pull request #13349 from ClickHouse/backport/20.4/13234

Backport #13234 to 20.4: Return passed number for numbers with MSB set in roundUpToPowerOfTwoOrZero()

CurtizJ committed:
Merge pull request #13480 from ClickHouse/backport/20.4/13396

Backport #13396 to 20.4: Fix assertion in FinishSortingTransform.

Commit:
Backport #13082 to 20.4: Fix access to redis dictionary after connection was dropped once

Commit:
Backport #13416 to 20.4: Possibly fix race in StorageMemory.

KochetovNicolai committed:
Merge pull request #13484 from ClickHouse/backport/20.4/13420

Backport #13420 to 20.4: Fix tuple with null comparison

KochetovNicolai committed:
Merge pull request #13568 from ClickHouse/backport/20.4/13416

Backport #13416 to 20.4: Possibly fix race in StorageMemory.

Commit:
Backport #13199 to 20.4: Fix parsing row policies from users.xml

Commit:
Merge pull request #13588 from ClickHouse/backport/20.4/13199

Backport #13199 to 20.4: Fix parsing row policies from users.xml

Commit:
Backport #13626 to 20.4: Fix deadlock and logical error in replication queue

akuzm committed:
Fix the 01428_hash_set_nan_key test not to rely on backwards-incompatible changes in `modulo` made in #12585

tavplubix committed:
Merge pull request #13670 from ClickHouse/backport/20.4/13626

Backport #13626 to 20.4: Fix deadlock and logical error in replication queue

CurtizJ committed:
Merge pull request #13558 from ClickHouse/backport/20.4/13082

Backport #13082 to 20.4: Fix access to redis dictionary after connection was dropped once

Commit:
Backport #13711 to 20.4: passwd and group location error

Commit:
Backport #13737 to 20.4: Fix logging Settings.Names/Values when log_queries_min_type > QUERY_START

Commit:
Backport #13790 to 20.4: Fix step overflow in range()

Commit:
Backport #13793 to 20.4: Fix set index with const column pred

KochetovNicolai committed:
Merge pull request #13731 from ClickHouse/backport/20.4/13711

Backport #13711 to 20.4: passwd and group location error

Commit:
Backport #13842 to 20.4: Fix lgamma race.

KochetovNicolai committed:
Merge pull request #13921 from ClickHouse/backport/20.4/13842

Backport #13842 to 20.4: Fix lgamma race.

Commit:
Auto version update to [20.4.9.110] [54434]

Commit:
Auto version update to [20.4.10.1] [54434]

alexey-milovidov committed:
Merge pull request #13811 from ClickHouse/backport/20.4/13793

Backport #13793 to 20.4: Fix set index with const column pred

alexey-milovidov committed:
Merge pull request #13802 from ClickHouse/backport/20.4/13790

Backport #13790 to 20.4: Fix step overflow in range()

alexey-milovidov committed:
Merge pull request #13775 from ClickHouse/backport/20.4/13737

Backport #13737 to 20.4: Fix logging Settings.Names/Values when log_queries_min_type > QUERY_START

Commit:
Backport #13922 to 20.4: Fix lock order inversion when renaming Distributed table

alexey-milovidov committed:
Merge pull request #13976 from ClickHouse/backport/20.4/13922

Backport #13922 to 20.4: Fix lock order inversion when renaming Distributed table

Commit:
Backport #14050 to 20.4: Fix \"no space left\" extra info for Poco::Exception

tavplubix committed:
Merge pull request #14075 from ClickHouse/backport/20.4/14050

Backport #14050 to 20.4: Fix "no space left" extra info for Poco::Exception

Commit:
Backport #14129 to 20.4: Slightly optimize very short queries with LowCardinality

CurtizJ committed:
Merge pull request #14186 from ClickHouse/backport/20.4/14129

Backport #14129 to 20.4: Slightly optimize very short queries with LowCardinality
